File: src/exemplos/03_Fluxo/02_Repeticao/04-do-while.py
#  -*- coding: utf-8 -*-
##   @package: 04-do-while.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Em Python, não há o laço de repetição DO-WHILE. Mas o
# comportamento pode ser replicado da seguinte forma:
#
# while True:
#     # iteração
#
#     if criterio_de_parada:
#         break


N = 10

print('do-while(i < %d)' % N)
print('\t', end=' ')
i = 0  # Inicialização do critério de parada.
while True:
    print(i, end=' ')
    i += 1  # Atualização da condição de parada.

    if i >= N:
        break

print()

File: src/exemplos/03_Fluxo/02_Repeticao/04-do-while.c
/**      @file: 04-do-while.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo do laço de repetição DO-WHILE. */

#include <stdio.h>

int main (){
    int i, N = 10;

    printf("do-while(i < %d)\n\t",N);
    i = 0; /* Inicialização do critério de parada. */
    do {
        printf("%d ", i++); /* Atualização da condição de parada. */
    } while(i < N);

    printf("\n");

    return 0; /* Não houve erro. */
}